### PR TITLE
[docs] torch.min/max: remove incorrect warning from docs

### LINKED ISSUE #44195: Remove torch.max/min warning
## 📚 Documentation

https://github.com/pytorch/pytorch/blob/f8f35fddd4b2bd788953d6d6ccfadf690b73a3e8/torch/_torch_docs.py#L4273-L4286

The following warning can be removed given that https://github.com/pytorch/pytorch/pull/42004 is in.

cc: @mruberry 

cc @jlin27 @mruberry @VitalyFedyunin

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes #44195

cc: @mruberry

### CODE REVIEW

**Comment by mruberry** [comment]
Should this also add a sentence to the first paragraph of the min/max documentation saying that if there are multiple minima/maxima then the index of the FIRST minima/maxima will be returned?

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -4332,6 +4332,8 @@ def merge_dicts(*dicts):
 .. warning::
     This function produces deterministic (sub)gradients unlike ``max(dim=0)``
 
+.. note:: If there are multiple minimal values then the indices of the first minimal value are returned.
```
Comment (mruberry) [inline torch/_torch_docs.py:4335]: This needs to say "maximal", not "minimal"

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -4350,13 +4352,6 @@ def merge_dicts(*dicts):
 :attr:`dim`. And ``indices`` is the index location of each maximum value found
 (argmax).
 
-.. warning::
```
Comment (mruberry) [inline torch/_torch_docs.py:4353]: The note actually needs to go with this version of max, which returns indices, not the first version which does not.

Further, the note should be slightly expanded, "If there are multiple maximal values in a reduced row then the indices of the first maximal value is returned." 

Parallel comments for the note below, too.

**Comment by kshitij12345** [comment]
@mruberry PTAL:)

**Review by mruberry — APPROVED** [review]
Excellent! Thanks @kshitij12345!

**Comment by kshitij12345** [comment]
@mruberry Thanks for the fix!!

**Pull request closed without merging**